concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -453,6 +493,85 @@ extern "C" {
+
+    pub fn freeByteArray(hstring: *const u8);
+
+    // Functions related to V2 GRPC interface.
+
+    /// Get information about a specific account in a given block.
+    pub fn getAccountInfoV2(
+        consensus: *mut consensus_runner,
+        acc_type: u8,
+        acc_id: *const u8,
+        block_id_type: u8,
+        block_hash: *const u8,
+        out_hash: *mut u8,
+        out: *mut Vec<u8>,
+        copier: CopyToVecCallback,
+    ) -> i64;
+
+    /// Get the list of accounts in a given block and, if the block exists,
+    /// enqueue them into the provided [Sender](futures::channel::mpsc::Sender).
+    ///
+    /// Individual account addresses are enqueued using the provided callback.
+    pub fn getAccountListV2(
+        consensus: *mut consensus_runner,
+        sender: *mut futures::channel::mpsc::Sender<Result<Vec<u8>, tonic::Status>>,
+        block_id_type: u8,
+        block_hash: *const u8,
+        out_hash: *mut u8,
+        callback: extern "C" fn(
+            *mut futures::channel::mpsc::Sender<Result<Vec<u8>, tonic::Status>>,
+            *const u8,
+            i64,
+        ) -> i32,
+    ) -> i64;
+}
+
+/// This is the callback invoked by consensus on newly arrived, and newly
+/// finalized blocks. The callback enqueues the block in the corresponding
+/// channel for further processing. The other end of the channel is owned by a
+/// background task that forwards data from the channels into any currently
+/// active RPC clients.
+unsafe extern "C" fn notify_callback(
+    notify_context: *mut NotificationContext,
+    ty: u8,
+    data_ptr: *const u8,
+    data_len: u64,
+) {
+    let sender = &*notify_context;
+    match ty {
+        0u8 => {
+            if sender
+                .blocks
+                .unbounded_send(std::slice::from_raw_parts(data_ptr, data_len as usize).into())
+                .is_err()
+            {
+                error!("Failed to enqueue block that arrived.");
+                // do nothing. The error here should only happen if the
+                // receiver is disconnected, which means that the task
+                // forwarding events has been killed. That should never happen,
+                // and if it does indicates a disastrous situation.
+            }
+        }
+        1u8 => {
+            if sender
+                .finalized_blocks
+                .unbounded_send(std::slice::from_raw_parts(data_ptr, data_len as usize).into())
+                .is_err()
+            {
+                error!("Failed to enqueue finalized block.");
+                // do nothing. The error here should only happen if the
+                // receiver is disconnected, which means that the task
+                // forwarding events has been killed. That should never happen,
+                // and if it does indicates a disastrous situation.
+            }
+        }
+        unexpected => {
+            error!("Unexpected notification type {}. This is a bug.", unexpected);
```

Comment:
I think 'Unexpected notification type {}' is sufficient :-). Otherwise we could be putting 'This is a bug' many places.

Reply:
I disagree. This is a bug, whereas other "error"s we have are not of this type. They are something that happens because of the environment, e.g., peer disconnected or something.